### PR TITLE
[#1183] Migrate multiwallet config; Refactor multiwallet config handling

### PR DESCRIPTION
**Pull request by placer14**
- Reconcile schema.DefaultWalletsConfig()
- Migrate legacy "Wallet" config key to "LegacyWallet" which is ignored
by config handling
- Migrate default "Wallets" config key to contain default multiwallet
settings. Ignores prior fee defaults, but legacy config is intact should
the user decide to restore their prior settings.
- A failure to create a single wallet will no longer impact other
wallets
- Removal of the coin key under the "Wallets" key will disable that
wallet. Alternatively, changing the Type to a value other than "API" and
"SPV" will also disable that coin.

### CODE REVIEW

File: repo/migrations/Migration015.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		return fmt.Errorf("reading config: %s", err.Error())
+	}
+
+	if err = json.Unmarshal(configBytes, &configMap); err != nil {
+		return fmt.Errorf("unmarshal config: %s", err.Error())
+	}
+
+	configMap["LegacyWallet"] = configMap["Wallet"]
+	delete(configMap, "Wallet")
+	configMap["Wallets"] = schema.DefaultWalletsConfig()
```
Comment (placer14) [inline repo/migrations/Migration015.go]: I need to remove this reference and use a literal instead.

File: wallet/builder.go [hunk 2]
```diff
@@ -2,247 +2,364 @@ package wallet
...
+		}
+	case wallet.BitcoinCash:
+		if notMainnet {
+			actualCoin = wallet.TestnetBitcoinCash
 		} else {
-			mw[wallet.BitcoinCash] = bitcoinCashSPVWallet
+			actualCoin = wallet.BitcoinCash
 		}
 	}
+	newSPVWallet, err := spvwallet.NewSPVWallet(coinConfig)
```
Comment (cpacia) [inline wallet/builder.go]: This will need to use `bitcoincash.NewSPVWallet()` if it's a bitcoin cash wallet.

File: schema/configuration.go [hunk 3]
```diff
@@ -57,6 +57,67 @@ type DataSharing struct {
...
+			MediumFeeDefault: 10,
+			HighFeeDefault:   50,
+			MaxFee:           200,
+			WalletOptions:    nil,
+		},
+		BCH: &CoinConfig{
+			Type:             WalletTypeAPI,
+			APIPool:          CoinPoolBCH,
+			APITestnetPool:   CoinPoolTBCH,
+			FeeAPI:           feeAPI,
```
Comment (cpacia) [inline schema/configuration.go]: I think `feeAPI` for everything other than BTC should bet set to "". If it isn't the wallet may try to use that API.

File: repo/migrations/Migration015.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+			MediumFeeDefault: 10,
+			HighFeeDefault:   50,
+			MaxFee:           200,
+			WalletOptions:    nil,
+		},
+		BCH: &migration015CoinConfig{
+			Type:             "API",
+			APIPool:          []string{"https://bch.api.openbazaar.org/api"},
+			APITestnetPool:   []string{"https://tbch.api.openbazaar.org/api"},
+			FeeAPI:           feeAPI,
```
Comment (cpacia) [inline repo/migrations/Migration015.go]: Same comment about the feeAPI as below

**Comment by cpacia** [comment]
This LGTM

**Pull request merged**